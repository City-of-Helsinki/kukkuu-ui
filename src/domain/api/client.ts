import ApolloClient from 'apollo-boost';

import { store } from '../app/state/AppStore';
import { apiTokenSelector } from '../auth/state/AuthenticationSelectors';

export default new ApolloClient({
  request: async operation => {
    try {
      const token = apiTokenSelector(store.getState());
      operation.setContext({
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
    } catch (e) {
      // User not authenticated
      // eslint-disable-next-line no-console
      console.error(e);
      // TODO: add error-handler
    }
  },
  uri: process.env.REACT_APP_API_URI,
});
