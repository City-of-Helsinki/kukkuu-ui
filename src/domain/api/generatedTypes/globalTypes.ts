/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

/**
 * An enumeration.
 */
export enum Language {
  EN = "EN",
  FI = "FI",
  SV = "SV",
}

export enum RelationshipTypeEnum {
  ADVOCATE = "ADVOCATE",
  OTHER_GUARDIAN = "OTHER_GUARDIAN",
  OTHER_RELATION = "OTHER_RELATION",
  PARENT = "PARENT",
}

export interface ChildInput {
  firstName?: string | null;
  lastName?: string | null;
  birthdate: any;
  postalCode?: string | null;
  relationship?: RelationshipInput | null;
}

export interface RelationshipInput {
  type?: RelationshipTypeEnum | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
