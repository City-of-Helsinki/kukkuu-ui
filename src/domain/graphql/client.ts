import ApolloClient from 'apollo-boost';

import getAuthenticatedUser from '../auth/getAuthenticatedUser';

export default new ApolloClient({
  request: async operation => {
    try {
      const user = await getAuthenticatedUser();
      operation.setContext({
        headers: {
          Authorization: `Bearer ${user.id_token}`,
        },
      });
    } catch (e) {
      // User not authenticated
      console.error(e);
    }
  },
  uri: process.env.REACT_APP_PROFILE_GRAPHQL,
});
