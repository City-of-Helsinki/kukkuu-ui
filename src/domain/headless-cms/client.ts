import {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
  ApolloLink,
} from '@apollo/client';
import { onError } from '@apollo/client/link/error';
import * as Sentry from '@sentry/browser';

const httpLink = createHttpLink({
  uri: process.env.REACT_APP_CMS_URI,
});

const errorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    graphQLErrors.forEach(({ message, locations, path, extensions }) => {
      const errorCode = extensions?.code;

      // eslint-disable-next-line max-len
      const errorMessage = `[GraphQL error]: Message: ${message}, Code: ${errorCode}, Location: ${locations}, Path: ${path}`;

      Sentry.captureMessage(errorMessage);

      if (process.env.NODE_ENV === 'development') {
        // eslint-disable-next-line no-console
        console.error(errorMessage);
      }
    });
  }

  if (networkError) {
    Sentry.captureMessage('Network error');
  }
});

const client = new ApolloClient({
  link: ApolloLink.from([errorLink, httpLink]),
  cache: new InMemoryCache(),
});

export default client;
