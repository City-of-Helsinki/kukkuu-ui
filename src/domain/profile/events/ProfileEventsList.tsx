import React, { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import { QRCode } from 'react-qrcode-logo';

import Card from '../../../common/components/card/Card';
import {
  childByIdQuery_child_availableEvents as AvailableEventsTypes,
  childByIdQuery_child_enrolments as EnrolmentsTypes,
  childByIdQuery_child_pastEvents as PastEventsTypes,
} from '../../api/generatedTypes/childByIdQuery';
import styles from './profileEventsList.module.scss';
import OccurrenceInfo from '../../event/partial/OccurrenceInfo';

interface ProfileEventsListProps {
  availableEvents: AvailableEventsTypes | null;
  childId: string;
  enrolments: EnrolmentsTypes;
  pastEvents: PastEventsTypes | null;
}

const QR_CODE_SIZE_PX = 180;

const ProfileEventsList: FunctionComponent<ProfileEventsListProps> = ({
  availableEvents,
  childId,
  enrolments,
  pastEvents,
}) => {
  const history = useHistory();
  const { t } = useTranslation();

  const gotoEventPage = (eventId: string) => {
    history.push(`/profile/child/${childId}/event/${eventId}`);
  };

  return (
    <>
      {availableEvents?.edges?.[0] && (
        <div className={styles.eventsList}>
          <h2>{t('profile.events.invitations.heading')}</h2>
          {availableEvents.edges.map(
            eventEdge =>
              eventEdge?.node && (
                <Card
                  key={eventEdge.node.id}
                  imageSrc={eventEdge.node.image}
                  title={eventEdge.node.name || ''} // TODO
                  action={() => gotoEventPage(eventEdge.node?.id || '')} // TODO
                  actionText={t('enrollment.enroll.buttonText')}
                  primaryAction={() => gotoEventPage(eventEdge.node?.id || '')} // TODO
                  primaryActionText={t('enrollment.enroll.buttonText')}
                >
                  <p>{eventEdge.node.shortDescription}</p>
                </Card>
              )
          )}
        </div>
      )}
      {enrolments.edges?.[0] && (
        <div className={styles.eventsList}>
          <h2>{t('profile.events.upcoming.heading')}</h2>
          {enrolments.edges.map(
            enrolmentEdge =>
              enrolmentEdge?.node?.occurrence && (
                <Card
                  key={enrolmentEdge.node.occurrence.event.id}
                  title={enrolmentEdge.node.occurrence.event.name || ''}
                  imageElement={
                    <div className={styles.qrWrapper}>
                      <QRCode
                        quietZone={0}
                        size={QR_CODE_SIZE_PX}
                        value={'Hello World - this works'}
                        ecLevel={'H'}
                      />
                    </div>
                  }
                  action={() =>
                    gotoEventPage(enrolmentEdge.node?.occurrence.event.id || '')
                  }
                  actionText={t('enrollment.showEventInfo.buttonText')}
                  focalContent={OccurrenceInfo(enrolmentEdge.node)}
                >
                  <p>{enrolmentEdge.node.occurrence.event.shortDescription}</p>
                </Card>
              )
          )}
        </div>
      )}
      {pastEvents?.edges?.[0] && (
        <div className={styles.eventsList}>
          <h2>{t('profile.events.past.heading')}</h2>
          {pastEvents.edges.map(
            pastEventEdge =>
              pastEventEdge?.node && (
                <Card
                  key={pastEventEdge.node.id}
                  imageSrc={pastEventEdge.node.image}
                  title={pastEventEdge.node.name || ''}
                  action={() => gotoEventPage(pastEventEdge.node?.id || '')}
                  actionText={t('enrollment.showEventInfo.buttonText')}
                >
                  <p>{pastEventEdge.node.shortDescription}</p>
                </Card>
              )
          )}
        </div>
      )}
    </>
  );
};

export default ProfileEventsList;
