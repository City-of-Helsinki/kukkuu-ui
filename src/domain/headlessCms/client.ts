import { ApolloClient, createHttpLink, ApolloLink } from '@apollo/client';
import { onError } from '@apollo/client/link/error';
import * as Sentry from '@sentry/browser';

import AppConfig from '../app/AppConfig';
import {
  createApolloCache,
  isConfiguredQueryPolicy,
  queryPolicies,
} from './cache';
import { LogLevel, TimedApolloCachePersistor } from './persistor';

export function createCmsApolloClient() {
  const httpLink = createHttpLink({
    uri: import.meta.env.VITE_CMS_URI,
  });

  const errorLink = onError(({ graphQLErrors, networkError }) => {
    if (graphQLErrors) {
      graphQLErrors.forEach(({ message, locations, path, extensions }) => {
        const errorCode = extensions?.code;

        // eslint-disable-next-line max-len
        const errorMessage = `[GraphQL error]: Message: ${message}, Code: ${errorCode}, Location: ${locations}, Path: ${path}`;

        Sentry.captureMessage(errorMessage);

        if (!AppConfig.isAppInProductionMode) {
          // eslint-disable-next-line no-console
          console.error(errorMessage);
        }
      });
    }

    if (networkError) {
      Sentry.captureMessage('Network error');
    }
  });

  const cache = createApolloCache();
  const kukkuuApolloCachePersistor = new TimedApolloCachePersistor(cache, {
    logLevel: LogLevel.WARN,
    persistedCacheTimeToLiveMs: AppConfig.apolloPersistedCacheTimeToLiveMs,
  });

  (async () => {
    if (kukkuuApolloCachePersistor.hasPersistedCacheExpired()) {
      // eslint-disable-next-line no-console
      console.info('Persisted cache has expired.');
      await kukkuuApolloCachePersistor.purge();
    } else {
      await kukkuuApolloCachePersistor.restore();
      // eslint-disable-next-line no-console
      console.info('Persisted cache has been restored.');
    }
  })();

  const client = new ApolloClient({
    link: ApolloLink.from([errorLink, httpLink]),
    cache,
    defaultOptions: {
      watchQuery: {
        nextFetchPolicy(
          currentFetchPolicy,
          {
            // Either "after-fetch" or "variables-changed", indicating why the
            // nextFetchPolicy function was invoked.
            reason,
            // The rest of the options (currentFetchPolicy === options.fetchPolicy).
            // The original value of options.fetchPolicy, before nextFetchPolicy was
            // applied for the first time.
            initialFetchPolicy,
            // The ObservableQuery associated with this client.watchQuery call.
            observable,
          }
        ) {
          // When variables change, the default behavior is to reset
          // options.fetchPolicy to context.initialFetchPolicy. If you omit this logic,
          // your nextFetchPolicy function can override this default behavior to
          // prevent options.fetchPolicy from changing in this case.
          if (reason === 'variables-changed') {
            return initialFetchPolicy;
          }
          // The languages are hardcoded in the cache, so we can always return cache-only,
          // because the languages never changes. Also, flashing on menus should be avoided,
          // so the menu query should first try from cache, but could still update cache
          // on background.
          if (isConfiguredQueryPolicy(observable.queryName)) {
            return queryPolicies[observable.queryName];
          }
          // Leave all other fetch policies unchanged.
          return currentFetchPolicy;
        },
      },
    },
  });

  return client;
}

const client = createCmsApolloClient();

export default client;
