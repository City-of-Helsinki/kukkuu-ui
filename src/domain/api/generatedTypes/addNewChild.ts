/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AddChildMutationInput } from './globalTypes';

// ====================================================
// GraphQL mutation operation: addNewChild
// ====================================================

export interface addNewChild_addChild_child_project {
  /**
   * The ID of the object.
   */
  id: string;
  name: string | null;
  year: number;
}

export interface addNewChild_addChild_child {
  /**
   * The ID of the object.
   */
  id: string;
  firstName: string;
  lastName: string;
  birthdate: any;
  postalCode: string;
  project: addNewChild_addChild_child_project;
}

export interface addNewChild_addChild {
  child: addNewChild_addChild_child | null;
}

export interface addNewChild {
  /**
   * This mutation cannot be used before one has started using the service with "SubmitChildrenAndGuardianMutation".
   */
  addChild: addNewChild_addChild | null;
}

export interface addNewChildVariables {
  input: AddChildMutationInput;
}
