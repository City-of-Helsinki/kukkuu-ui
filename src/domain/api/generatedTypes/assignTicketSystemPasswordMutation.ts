/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssignTicketSystemPasswordMutationInput } from "./globalTypes";

// ====================================================
// GraphQL mutation operation: assignTicketSystemPasswordMutation
// ====================================================

export interface assignTicketSystemPasswordMutation_assignTicketSystemPassword {
  /**
   * The assigned ticket system password
   */
  password: string | null;
}

export interface assignTicketSystemPasswordMutation {
  assignTicketSystemPassword: assignTicketSystemPasswordMutation_assignTicketSystemPassword | null;
}

export interface assignTicketSystemPasswordMutationVariables {
  input: AssignTicketSystemPasswordMutationInput;
}
