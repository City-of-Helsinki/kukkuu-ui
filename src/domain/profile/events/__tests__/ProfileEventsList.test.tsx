import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import * as reactQrCodeLogo from 'react-qrcode-logo';

import { render, screen } from '../../../../common/test/testingLibraryUtils';
import ProfileEventsList from '../ProfileEventsList';
import {
  childByIdQuery_child as ChildByIdResponse,
  childByIdQuery_child_upcomingEventsAndEventGroups as UpcomingEventsAndEventGroupsType,
  childByIdQuery_child_pastEvents as PastEventsType,
  childByIdQuery_child_activeInternalAndTicketSystemEnrolments as InternalAndTicketSystemEnrolmentsType,
} from '../../../api/generatedTypes/childByIdQuery';
import { EventParticipantsPerInvite } from '../../../api/generatedTypes/globalTypes';

vi.spyOn(reactQrCodeLogo, 'QRCode').mockImplementation(() => (<div />) as any);

const childData: ChildByIdResponse = {
  id: '',
  firstName: '',
  lastName: '',
  birthdate: '',
  postalCode: '',
  project: {
    id: '',
    year: 0,
    name: 'Test project',
  },
  relationships: { edges: [] },
  upcomingEventsAndEventGroups: { edges: [] },
  activeInternalAndTicketSystemEnrolments: { edges: [] },
  pastEvents: { edges: [] },
};

const eventData = {
  id: 'RXZlbnROb2RlOjE=',
  name: 'pentti',
  shortDescription: 'eventti',
  image: 'http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png',
  imageAltText: 'huhuu',
  duration: 60,
  participantsPerInvite: EventParticipantsPerInvite.CHILD_AND_GUARDIAN,
  occurrences: { edges: [] },
  canChildEnroll: true,
};

const upcomingEventsAndEventGroups: UpcomingEventsAndEventGroupsType = {
  edges: [
    {
      node: { ...eventData, __typename: 'EventNode' },
    },
  ],
};

const venueData = {
  id: 'uuap',
  name: 'aa',
  description: 'zzww',
  address: 'ssfas uus 12',
};

const enrolments: InternalAndTicketSystemEnrolmentsType = {
  edges: [
    {
      node: {
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        __typename: 'EnrolmentNode',
        id: 'foo',
        referenceId: 'bar',
        occurrence: {
          id: '',
          time: '2020-02-24T07:07:18+00:00', // 09.07
          venue: venueData,
          event: eventData,
        },
      },
    },
  ],
};

const pastEvents: PastEventsType = {
  edges: [
    {
      node: eventData,
    },
  ],
};

const childWithEvents: ChildByIdResponse = {
  ...childData,
  upcomingEventsAndEventGroups,
  activeInternalAndTicketSystemEnrolments: enrolments,
  pastEvents: pastEvents,
};

const childOnlyAvailableEvents = {
  ...childData,
  upcomingEventsAndEventGroups,
  enrolments: {
    edges: [],
  },
  pastEvents: null,
};

const childOnlyEnrolments: ChildByIdResponse = {
  ...childData,
  upcomingEventsAndEventGroups: null,
  activeInternalAndTicketSystemEnrolments: {
    edges: [
      {
        node: {
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          __typename: 'EnrolmentNode',
          id: 'foo',
          referenceId: 'bar',
          occurrence: {
            id: 'uu',
            time: '2020-02-24T09:09:09+00:00',
            venue: venueData,
            event: eventData,
          },
        },
      },
    ],
  },
  pastEvents: null,
};

const childWithTicketmasterEnrolment: ChildByIdResponse = {
  ...childData,
  upcomingEventsAndEventGroups: null,
  activeInternalAndTicketSystemEnrolments: {
    edges: [
      {
        node: {
          __typename: 'TicketmasterEnrolmentNode',
          id: 'ticketmaster-123',
          event: eventData,
        },
      },
    ],
  },
  pastEvents: null,
};

const childWithLippupisteEnrolment: ChildByIdResponse = {
  ...childData,
  upcomingEventsAndEventGroups: null,
  activeInternalAndTicketSystemEnrolments: {
    edges: [
      {
        node: {
          __typename: 'LippupisteEnrolmentNode',
          id: 'lippupiste-123',
          event: eventData,
        },
      },
    ],
  },
  pastEvents: null,
};

test('Renders snapshot correctly', () => {
  const { container } = render(
    <MockedProvider>
      <ProfileEventsList
        upcomingEventsAndEventGroups={
          childWithEvents.upcomingEventsAndEventGroups
        }
        enrolments={childOnlyEnrolments.activeInternalAndTicketSystemEnrolments}
        pastEvents={childWithEvents.pastEvents}
        childId="zzaf"
      />
    </MockedProvider>
  );
  expect(container).toMatchSnapshot();
});

test('Renders only upcoming events and event groups when no other inputs', () => {
  render(
    <MockedProvider>
      <ProfileEventsList
        upcomingEventsAndEventGroups={
          childOnlyAvailableEvents.upcomingEventsAndEventGroups
        }
        enrolments={
          childOnlyAvailableEvents.activeInternalAndTicketSystemEnrolments
        }
        pastEvents={childOnlyAvailableEvents.pastEvents}
        childId="zzaf"
      />
    </MockedProvider>
  );
  expect(
    screen.getByRole('heading', { name: 'Tapahtumakutsut' })
  ).toBeInTheDocument();
  expect(
    screen.queryByRole('heading', { name: 'Tulevat tapahtumasi' })
  ).not.toBeInTheDocument();
  expect(
    screen.queryByRole('heading', { name: 'Menneet tapahtumat' })
  ).not.toBeInTheDocument();
});

test('Renders only enrolments when no other inputs', () => {
  render(
    <MockedProvider>
      <ProfileEventsList
        upcomingEventsAndEventGroups={
          childOnlyEnrolments.upcomingEventsAndEventGroups
        }
        enrolments={childOnlyEnrolments.activeInternalAndTicketSystemEnrolments}
        pastEvents={childOnlyEnrolments.pastEvents}
        childId="zzaf"
      />
    </MockedProvider>
  );

  expect(
    screen.queryByRole('heading', { name: 'Tapahtumakutsut' })
  ).not.toBeInTheDocument();
  expect(
    screen.getByRole('heading', { name: 'Tulevat tapahtumasi' })
  ).toBeInTheDocument();
  expect(
    screen.queryByRole('heading', { name: 'Menneet tapahtumat' })
  ).not.toBeInTheDocument();
});

test.each([
  [
    'Ticketmaster',
    childWithTicketmasterEnrolment.activeInternalAndTicketSystemEnrolments,
  ],
  [
    'Lippupiste',
    childWithLippupisteEnrolment.activeInternalAndTicketSystemEnrolments,
  ],
])('Renders %s enrolment', (_, enrolments) => {
  render(
    <MockedProvider>
      <ProfileEventsList
        upcomingEventsAndEventGroups={{
          edges: [],
        }}
        enrolments={enrolments}
        pastEvents={{
          edges: [],
        }}
        childId="zzaf"
      />
    </MockedProvider>
  );

  expect(screen.getByText('eventti')).toBeInTheDocument();
});
