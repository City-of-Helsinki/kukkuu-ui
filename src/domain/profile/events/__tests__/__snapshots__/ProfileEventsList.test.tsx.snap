// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders snapshot correctly 1`] = `
<div>
  <ul
    class="_list_a90dcd _spacing-xl_a90dcd"
  >
    <li
      class="_item_a90dcd"
    >
      <h2
        class="_text_e4e173 _h2_e4e173 "
      >
        Tulevat tapahtumasi
      </h2>
      <ul
        class="_list_a90dcd _spacing-layout-2-xs_a90dcd"
      >
        <li
          class="_item_a90dcd"
        >
          <div
            class="_wrapper_f3c708"
          >
            <div
              class="_image_f3c708"
            >
              <div
                class="_qrWrapper_e3c7ac"
              >
                <div />
              </div>
            </div>
            <div
              class="_content_f3c708"
            >
              <h3
                class="_text_e4e173 _h3_e4e173 _title_f3c708"
              >
                pentti
              </h3>
              <p
                class="_text_e4e173 _body_e4e173 "
              >
                eventti
              </p>
              <div
                class="_focalPoint_f3c708"
              >
                <span>
                  <div
                    class="_row_e93b69"
                  >
                    <div
                      class="_label_e93b69 _fullWidth_e93b69"
                      id="venue"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M11.967 1.5c2.06 0 4.12.778 5.69 2.334 3.143 3.111 2.93 7.96 0 11.268l-.622.709c-2.612 2.991-4.066 4.96-5.068 6.937-1.073-2.13-2.682-4.249-5.689-7.646-2.93-3.308-3.143-8.157 0-11.268A8.06 8.06 0 0111.967 1.5zm.032 2a6.072 6.072 0 00-4.3 1.762A5.606 5.606 0 006.002 9.41c.02 1.573.648 3.134 1.766 4.398l.66.752c1.59 1.823 2.717 3.239 3.573 4.503.975-1.437 2.292-3.063 4.233-5.255 1.118-1.264 1.746-2.825 1.766-4.398a5.616 5.616 0 00-1.698-4.15A6.077 6.077 0 0011.999 3.5zM12 6a3.5 3.5 0 110 6.999A3.5 3.5 0 0112 6zm0 2c-.827 0-1.5.673-1.5 1.5S11.173 11 12 11s1.5-.673 1.5-1.5S12.827 8 12 8z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="_text_e4e173 _body_e4e173 _label_e93b69"
                        >
                          aa
                        </p>
                        <p
                          class="_text_e4e173 _body_e4e173 _description_e93b69"
                        >
                          ssfas uus 12
                        </p>
                      </div>
                    </div>
                    <div
                      class="_label_e93b69"
                      id="time"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M17 2a1 1 0 011 1v1h4v17H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 9H4v8h16v-8zm0-5H4v3h16V6z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="_text_e4e173 _body_e4e173 _label_e93b69"
                        >
                          24.2.2020
                        </p>
                      </div>
                    </div>
                    <div
                      class="_label_e93b69"
                      id="duration"
                    >
                      <svg
                        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                        role="img"
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <g
                          fill="none"
                          fill-rule="evenodd"
                        >
                          <path
                            d="M0 0h24v24H0z"
                          />
                          <path
                            d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 2v5.584l3.243 3.244-1.415 1.415-3.535-3.536.002-.001-.294-.292L11 6h2z"
                            fill="currentColor"
                          />
                        </g>
                      </svg>
                      <div>
                        <p
                          class="_text_e4e173 _body_e4e173 _label_e93b69"
                        >
                          11:09 - 12:09
                        </p>
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </div>
            <div
              class="_cta_f3c708"
            >
              <button
                aria-label="Näytä tapahtuman tiedot"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO _actionWrapper_f3c708"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
    <li
      class="_item_a90dcd"
    >
      <h2
        class="_text_e4e173 _h2_e4e173 "
      >
        Tapahtumakutsut
      </h2>
      <ul
        class="_list_a90dcd _spacing-layout-2-xs_a90dcd"
      >
        <li
          class="_item_a90dcd"
        >
          <div
            class="_wrapper_f3c708"
          >
            <div
              class="_image_f3c708"
            >
              <img
                alt="huhuu"
                class="_image_f3c708"
                src="http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png"
              />
            </div>
            <div
              class="_content_f3c708"
            >
              <h3
                class="_text_e4e173 _h3_e4e173 _title_f3c708"
              >
                pentti
              </h3>
              <p
                class="_text_e4e173 _body_e4e173 "
              >
                eventti
              </p>
              <div
                class="_focalPoint_f3c708"
              >
                <button
                  class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO _primaryActionButton_f3c708"
                  style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: var(--color-summer); --background-color-hover: var(--color-summer-dark); --background-color-focus: var(--color-summer); --background-color-hover-focus: var(--color-summer-dark); --background-selected: var(--color-summer); --border-color: var(--color-summer); --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                  type="button"
                >
                  <span
                    class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                  >
                    Lue lisää
                  </span>
                </button>
                <span />
              </div>
            </div>
            <div
              class="_cta_f3c708"
            >
              <button
                aria-label="Lue lisää"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO _actionWrapper_f3c708"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
    <li
      class="_item_a90dcd"
    >
      <h2
        class="_text_e4e173 _h2_e4e173 "
      >
        Menneet tapahtumat
      </h2>
      <ul
        class="_list_a90dcd _spacing-layout-2-xs_a90dcd"
      >
        <li
          class="_item_a90dcd"
        >
          <div
            class="_wrapper_f3c708"
          >
            <div
              class="_image_f3c708"
            >
              <img
                alt="huhuu"
                class="_image_f3c708"
                src="http://localhost:8081/media/2020-02-15-184035_1920x1080_scrot.png"
              />
            </div>
            <div
              class="_content_f3c708"
            >
              <h3
                class="_text_e4e173 _h3_e4e173 _title_f3c708"
              >
                pentti
              </h3>
              <p
                class="_text_e4e173 _body_e4e173 "
              >
                eventti
              </p>
              <div
                class="_focalPoint_f3c708"
              >
                <span />
              </div>
            </div>
            <div
              class="_cta_f3c708"
            >
              <button
                aria-label="Näytä tapahtuman tiedot"
                class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO _actionWrapper_f3c708"
                style="--color: var(--color-black); --color-hover: var(--color-black); --color-focus: var(--color-black-90); --color-hover-focus: var(--color-black-90); --background-color: transparent; --background-color-hover: transparent; --background-color-focus: transparent; --background-color-hover-focus: transparent; --background-selected: var(--color-summer); --border-color: none; --border-color-hover: var(--color-summer-dark); --border-color-focus: var(--color-summer-dark); --border-color-hover-focus: var(--color-summer-dark); --border-color-selected: var(--color-summer); --border-color-selected-hover: var(--color-summer-dark); --border-color-selected-focus: var(--color-summer); --border-color-selected-hover-focus: var(--color-summer);"
                type="button"
              >
                <span
                  class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
                >
                  <svg
                    class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                    role="img"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="none"
                      fill-rule="evenodd"
                    >
                      <polygon
                        points="0 24 0 0 24 0 24 24"
                      />
                      <polygon
                        fill="currentColor"
                        points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
                      />
                    </g>
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
